.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,47 @@
+name: Publish Docs
+
+on:
+  push:
+    tags:
+      # Only runs for version tags:
+      # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-of-specific-tags-occurs
+      - '[0-9]+.[0-9]+.[0-9]+'
+
+permissions:
+  contents: write
```

Comment:
Where does this write to? The current repository? I'd strongly recommend to avoid giving the build pipeline write access to the source code.

Reply:
> Where does this write to?

Precisely for this workflow, it would write to `gh-pages` branch. I'm not happy with how extensive these permissions are either, but GitHub does not have a way to configure permissions as fine-grained as "push to gh-pages branch". Or at least I couldn't find how to do it. 

I'm not worried about this that much, though. This is a workflow running inside GitHub, not a third-party service. The workflow only changes when someone authorized to write to the `main` branch merges/pushes commits there. Additionally, the workflow is configured to only run for tags (in other words, when someone authorized pushes a new one).

Overall, I think automating this when tags are created makes it worth having such permission. One change I can make though is to move the permissions to the job (instead of being workflow-wide).

What do you think, @kelunik?

Reply:
> One change I can make though is to move the permissions to the job (instead of being workflow-wide).

See 88cfc4026ea00a51218d4592c9237f7fab3bb89a.

Reply:
Would it help to grant write permissions to a different repository, for instance jte-website, to publish the data?

Reply:
> Would it help to grant write permissions to a different repository, for instance jte-website, to publish the data?

It may be tricky to achieve this if the docs source lives here (the ideal place for them to be, IMO). But I see GitHub has a new (beta) feature to publish GH Pages from a workflow, and that would require fewer permissions.

I will try to find the time this week and see how that works. 

But, if this is blocking the PR, I would instead remove it for now. It is here more for the convenience of automatically publishing the docs whenever a release is made.

WDYT?

Reply:
@marcospereira I think we don't need to rush it. Having this as a separate workflow sounds like a good idea, because then we wouldn't have a branch with lots of changes compared to main and we would not need to give the tooling permission to alter this repository.

This would mean, that we wouldn't need mike at all right? And probably also have no version history?

Reply:
Sorry, @casid. I'm not sure I got what you said.

Do you mean remove the docs publishing workflow, versioning, and then have the publishing process as a manual step? If so, I'm okay with it too. And yes, that would mean `mike` wouldn't be necessary (which, tbh, simplifies things).

Reply:
Yes, I think this would mean to remove versioning and mike, which would simplify things.

I don't think we have to have a manual publishing process, I _think_ that it should be possible to adjust the current workflow to use this: https://docs.github.com/en/pages/getting-started-with-github-pages/using-custom-workflows-with-github-pages 

Looking at the workflow file I think it is already almost 90% done this way. I think we would only remove the ability to write contents in the jte repository, remove mike and pushing to a branch. Sorry in case I got this all wrong!

Reply:
You got it right. I was looking at the same docs you linked, and I'm adjusting the workflow now. I'm trying it on my fork first, hence moving the PR to draft for now. 🏃🏼‍♂️

Reply:
I saw your commit after posting my comment :-)

```
permissions:
  # Checkout the project's code
  contents: read
  # Allow to deploy to gh-pages
  pages: write
```

This looks like the tools are only allowed to push to the gh-pages branch? In that case we could keep mike and versioning, right?

Reply:
> This looks like the tools are only allowed to push to the gh-pages branch? In that case we could keep mike and versioning, right?

No. This means the actions are allowed to upload artifacts to GitHub Pages. But no write access to branches at all (including gh-pages). See my last commit.

Reply:
Ah, right. Thanks for giving it a try! Let's start with the simplified version. We can still add mike and versioning the day GitHub has more fine-grained permissions.

Let me know when you think this is ready for merging!

Reply:
Sure! For now, it looks like GH is not super happy about triggering the flow when a new tag is pushed:

https://github.com/marcospereira/jte/actions/runs/6569712256

🤨